stl/inc/vector

```diff
@@ -1730,25 +1765,43 @@ private:
                 *_Pnext             = (*_Pnext)->_Mynextiter;
             }
         }
+    }
+
+    void _Orphan_range_locked(pointer _First, pointer _Last) const {
+        _Lockit _Lock(_LOCK_DEBUG);
+        _Orphan_range_unlocked(_First, _Last);
+    }
+#endif // _ITERATOR_DEBUG_LEVEL == 2
+
+    _CONSTEXPR20_DYNALLOC void _Orphan_range(pointer _First, pointer _Last) const {
+#if _ITERATOR_DEBUG_LEVEL == 2
+#ifdef __cpp_lib_is_constant_evaluated
+        if (_STD is_constant_evaluated()) {
+            _Orphan_range_unlocked(_First, _Last);
+        } else
+#endif // __cpp_lib_is_constant_evaluated
+        {
+            _Orphan_range_locked(_First, _Last);
+        }
 #else // ^^^ _ITERATOR_DEBUG_LEVEL == 2 ^^^ // vvv _ITERATOR_DEBUG_LEVEL != 2 vvv
         (void) _First;
         (void) _Last;
 #endif // _ITERATOR_DEBUG_LEVEL == 2
     }

```

Comment:
I am wondering whether this would be cleaner with 
```suggestion
    _CONSTEXPR20_DYNALLOC void _Orphan_range(pointer _First, pointer _Last) const {
#ifdef __cpp_lib_is_constant_evaluated
        if (_STD is_constant_evaluated()) {
            _Orphan_range_unlocked(_First, _Last);
        } else
#endif // __cpp_lib_is_constant_evaluated
        {
            _Orphan_range_locked(_First, _Last);
        }
    }
#else // ^^^ _ITERATOR_DEBUG_LEVEL == 2 ^^^ // vvv _ITERATOR_DEBUG_LEVEL != 2 vvv
    _CONSTEXPR20_DYNALLOC void _Orphan_range(pointer, pointer) const {}
#endif // _ITERATOR_DEBUG_LEVEL == 2
    }
```

Reply:
Done that